### PR TITLE
Detached pgp verify was broken for messages shorter than 100 bytes

### PR DESCRIPTION
**Pull request by zanderz**
As per https://keybase.atlassian.net/browse/CORE-2267
`C:\Program Files\Keybase>keybase pgp verify -d sig1.txt -i msg1.txt`
`   ERROR unexpected EOF`
@maxtaco @oconnor663

### CODE REVIEW

File: go/libkb/stream_classifier.go [hunk 1]
```diff
@@ -53,7 +53,7 @@ func (p *StreamPeeker) Peek(buf []byte) (n int, err error) {
 		return 0, ErrCannotPeek
 	}
 	n, err = io.ReadFull(p.r, buf)
-	p.buf = append(p.buf, buf...)
+	p.buf = append(p.buf, buf[:n]...)
```
Comment (zanderz) [inline go/libkb/stream_classifier.go]: This was copying all 100 bytes of buf even if the message was shorter

File: packaging/windows/build_prerelease.cmd [hunk 2]
```diff
@@ -6,12 +7,12 @@ echo %KEYBASE_BUILD%
...
-go build -tags "production prerelease"
+::cd %GOPATH%\src\github.com\keybase\kbfs\kbfsdokan
+::go build -tags "production prerelease"
 
-Then the desktop:
-cd %GOPATH%\src\github.com\keybase\client\react-native\react
-npm i
-cd %GOPATH%\src\github.com\keybase\client\desktop
-npm i
-node package.js --arch ia32 --platform win32
```
Comment (zanderz) [inline packaging/windows/build_prerelease.cmd]: This stuff not related to the Jira ticket, but handy nonetheless

File: go/engine/pgp_verify_test.go [hunk 3]
```diff
@@ -74,6 +72,16 @@ func TestPGPVerify(t *testing.T) {
 	// verify that attached signature
 }
 
+func TestPGPVerify(t *testing.T) {
+	msg := "If you wish to stop receiving notifications from this topic, please click or visit the link below to unsubscribe:"
+	doVerify(t, msg)
+}
+
+func TestPGPVerifyShortMsg(t *testing.T) {
```
Comment (zanderz) [inline go/engine/pgp_verify_test.go]: I verified this new test failed before the included changes

File: go/libkb/stream_classifier.go [hunk 4]
```diff
@@ -235,7 +235,8 @@ func ClassifyStream(r io.Reader) (sc StreamClassification, out io.Reader, err er
 	peeker := NewStreamPeeker(r)
 	var buf [100]byte
 	var n int
-	if n, err = peeker.Peek(buf[:]); err != nil {
+	// ErrUnexpectedEOF might just mean we read less than 100 bytes
+	if n, err = peeker.Peek(buf[:]); err != nil && !(err == io.ErrUnexpectedEOF && len(buf) > 0) {
```
Comment (maxtaco) [inline go/libkb/stream_classifier.go]: don't you need to set `err = nil` in the case it was `io.ErrUnexpectedEOF`?

Comment (zanderz) [inline go/libkb/stream_classifier.go]: UnknownStreamError gets returned anyway in the short message detached case, but yeah, I need to do that for the other cases.

Comment (maxtaco) [inline go/libkb/stream_classifier.go]: Ah ok that makes sense.  But yeah you should clear out the error for the
case of short messages that are. Amid.

Comment (maxtaco) [inline go/libkb/stream_classifier.go]: Erm valid.

**Pull request merged**